controllers/cluster/upgrade_checker.go

```diff
@@ -52,13 +60,27 @@ func (o *openshiftClusterUpgradeStatusChecker) Check() (bool, error) {
 	return false, nil
 }
 
+func (o *openshiftClusterUpgradeStatusChecker) isNodeUpgrading(nodesToBeRemediated []corev1.Node) bool {
+	// We can identify an HCP (Hosted Control Plane) cluster upgrade by checking annotations of the CP node.
+	for _, node := range nodesToBeRemediated {
+		if !nodes.IsControlPlane(&node) || len(node.Annotations) == 0 || len(node.Annotations[desiredMachineConfigAnnotationKey]) == 0 {
```

Comment:
I might miss something, but why this check for cp node? :thinking:

Reply:
Probably originated in my mistake in the [original](https://redhat-internal.slack.com/archives/C03M5GKJNBA/p1740042464529109?thread_ts=1740042457.508119&cid=C03M5GKJNBA) (slack link) solution assumption.

>I want to outline my approach about how to go about it, to make sure I got the problem correctly and check whether my approach to solve it makes sense.
>1. pass [matchingNodes](https://github.com/medik8s/node-healthcheck-operator/blob/main/controllers/nodehealthcheck_controller.go#L248) to isClusterUpgrading
>2. in isClusterUpgrading filter for CP nodes and on those node check for annotation[CurrentMachineConfigAnnotationKey] != annotation[DesiredMachineConfigAnnotationKey] .
...

I'll updated the code.